src/pkg/services/m365/api/graph/service.go

```diff
@@ -442,6 +442,13 @@ func (aw *adapterWrap) Send(
 			// to limit the scope of this fix.
 			logger.Ctx(ictx).Debug("invalid request")
 			events.Inc(events.APICall, "invalidgetrequest")
+		} else if errors.Is(err, ErrNotFoundEmptyResp) {
```

Comment:
Sorry, this occurred to me just now. Should we restrict the retries to `GET` requests only. This will ensure we don't retry on `POST` 404s with empty content., as those can be non-idempotent. 

```suggestion
		} else if requestInfo.Method.String() == http.MethodGet && errors.Is(err, ErrNotFoundEmptyResp) {
```

Reply:
good callout!